webhook/webhook.groovy

```diff
@@ -161,7 +184,22 @@ storage {
      * item (org.artifactory.fs.ItemInfo) - the original item being created.
      */
     afterCreate { item ->
-        hook(Globals.SUPPORT_MATRIX.storage.afterCreate.name, item ? new JsonBuilder(item) : null)
+        if (WebHook.isEnableSpinnakerSupport()){
```

Comment:
Overall the changes look good to me and I have verified they run and return what is expected. This part here bothers me though, it means that if a user enabled Spinnaker support, it will alter the structure of all the other formats. Users will likely want to use this for Spinnaker and other formats as well. I recommend you revert the changes to afterCreate and instead move what you are doing here to somewhere else that will only change the structure to the Spinnaker format.

Reply:
I have to agree that this construction is a little weird. The structure should be similar enough that the other formats will work as expected (as written at the moment, anyway), but you're essentially removing a number of fields and adding one. Sure, if those other fields are necessary in the future they can be accessed through the `item` field, but that's only available if `enableSpinnakerSupport` is true, so the other formats wouldn't be able to depend on that.

Since it seems like the only thing you're getting out of this is the `type` field, and that can be derived at any time from the `repoKey`, this should be left as it was and the `type` should be calculated in the spinnaker-specific `format` method, right before it's needed. Then you won't need to change the structure at all.

Reply:
Thanks for your feedback and I agree with both of you @jfrog-aparicio, @DarthFennec.

Refactored code to move packageType retrival logic in the Spinnaker-specific format method and removed enableSpinnakerSupport. This allows using Spinnaker and other formats as well.